### PR TITLE
Add ChaCha20Poly1305

### PR DESCRIPTION
**Pull request by l1h3r**
# Description of change

Adds `ChaCha20Poly1305` for JSON web tokens used in identity.rs

## Type of change

- [x] Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

RFC test vectors

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by felsweg-iota — APPROVED** [review]
Looks good to me.

File: tests/fixtures/chacha20_poly1305.rs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[
+    // https://tools.ietf.org/html/rfc8439#section-2.8.2
```
Comment (semenov-vladyslav) [inline tests/fixtures/chacha20_poly1305.rs:2]: It's possible to extract another test vector from https://datatracker.ietf.org/doc/html/rfc8439#appendix-A

**Review by semenov-vladyslav — APPROVED** [review]
Looks good. There could be more test vectors.

**Comment by Wollac** [comment]
Seems like this PR is no longer needed. Please just re-open if it still is.

**Pull request closed without merging**